src/rust/engine/task_executor/src/lib.rs

```diff
@@ -119,7 +120,9 @@ impl Executor {
     // When a daemon thread kicks off a future, it should log like a daemon thread (and similarly
     // for a user-facing thread).
     self
-      .handle
+      .runtime
+      .as_ref()
+      .unwrap()
```

Comment:
This is returning `None` and causing some tests to fail.

Reply:
Mm. Yea, this is core to the `block_on` discussion, I think. The `handle` is a weak-reference to the `Runtime`, which doesn't prevent shutdown, and so it can fail.